packages/mojo-drupal-scaffold/assets/mojo.settings.php

```diff
@@ -0,0 +1,209 @@
+    'driver' => 'sqlite',
+  );
+}
+
+/**
+ * Email server connection information
+ */
+$config['swiftmailer.transport']['transport'] = 'smtp';
+$config['swiftmailer.transport']['smtp_host'] = $_ENV['SMTP_SERVER'];
+$config['swiftmailer.transport']['smtp_port'] = $_ENV['SMTP_PORT'];
+$config['swiftmailer.transport']['smtp_encryption'] = $_ENV['SMTP_PORT'] === '1025' ? '0' : 'ssl';
+$config['swiftmailer.transport']['smtp_credentials']['swiftmailer']['username'] = $_ENV['SMTP_USERNAME'];
+$config['swiftmailer.transport']['smtp_credentials']['swiftmailer']['password'] = $_ENV['SMTP_PASSWORD'];
+
+// Reverse proxy detection.
+// Stolen from the trusted_reverse_proxy module.
+if (isset($_SERVER['HTTP_X_FORWARDED_FOR'], $_SERVER['REMOTE_ADDR'])) {
+  $settings['reverse_proxy'] = TRUE;
+
+  // First hop is assumed to be a reverse proxy in its own right.
+  $proxies = [$_SERVER['REMOTE_ADDR']];
+  // We may be further behind another reverse proxy (e.g., Traefik, Varnish)
+  // Commas may or may not be followed by a space.
+  // @see https://tools.ietf.org/html/rfc7239#section-7.1
+  $forwardedFor = explode(
+    ',',
+    str_replace(', ', ',', $_SERVER['HTTP_X_FORWARDED_FOR'])
+  );
+  if (count($forwardedFor) > 1) {
+    // The first value will be the actual client IP.
+    array_shift($forwardedFor);
+    array_unshift($proxies, ...$forwardedFor);
+  }
+
+  $settings['reverse_proxy_addresses'] = $proxies;
+}
+
+if ($_ENV['FILESYSTEM_DRIVER'] === 's3') {
+  // We cannot support a private file path if using object storage.
+  unset($settings['file_private_path']);
+
+  $schemes = [
+    's3' => [
+      'driver' => 's3',
+      'config' => [
+        'key'    => $_ENV['AWS_ACCESS_KEY_ID'],
+        'secret' => $_ENV['AWS_SECRET_ACCESS_KEY'],
+        'region' => $_ENV['AWS_DEFAULT_REGION'],
+        'bucket' => $_ENV['S3_BUCKET'],
+        'endpoint' => $_ENV['S3_ENDPOINT'] ?? NULL,
+        'protocol' => $_ENV['S3_PROTOCOL'] ?? 'https',
+        'cname' => $_ENV['S3_CNAME'] ?? '',
+        'use_path_style_endpoint' => (bool) $_ENV['S3_USE_PATH_STYLE_ENDPOINT'],
+        'cname_is_bucket' => (bool) $_ENV['S3_CNAME_IS_BUCKET'],
+        'public' => TRUE,
+        'options' => [
+          # @todo is this needed?
+          'ACL' => 'public-read',
+        ],
+      ],
+      'cache' => TRUE,
+      // Without having this, the state value of `drupal_css_cache_files` and
+      // `system.js_cache_files` will have a mapping of non-existent files as
+      // disk storage is ephemeral in cloud environments.
+       'serve_js' => TRUE,
+       'serve_css' => TRUE,
+    ]
+  ];
+  $settings['flysystem'] = $schemes;
+  // Set the default scheme to be S3.
+  $config['system.file']['default_scheme'] = 's3';
+}
+
+if ($_ENV['FILESYSTEM_DRIVER'] !== 'local') {
+  // If the filesystem isn't local, move Twig to the temporary directory.
+  $settings['php_storage']['twig']['directory'] = $settings['file_temp_path'];
+}
```

Comment:
Just like there is `settings.redis.php` maybe anything for the filesystem should go in its own, as well.